src/dragonball/src/api/v1/virtio_net.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright (C) 2019-2023 Alibaba Cloud. All rights reserved.
+// Copyright (C) 2019-2023 Ant Group. All rights reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use dbs_utils::net::MacAddr;
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature = "virtio-net")]
+use super::{VirtioNetDeviceConfigInfo, VirtioNetDeviceConfigUpdateInfo};
+use crate::config_manager::RateLimiterConfigInfo;
+#[cfg(feature = "vhost-net")]
+use crate::device_manager::vhost_net_dev_mgr::{self, VhostNetDeviceConfigInfo};
+#[cfg(feature = "virtio-net")]
+use crate::device_manager::virtio_net_dev_mgr;
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
+#[serde(tag = "type", deny_unknown_fields)]
+/// An enum to specify a backend of Virtio network
+pub enum Backend {
+    #[serde(rename = "virtio")]
+    #[cfg(feature = "virtio-net")]
+    /// Virtio-net
+    Virtio(VirtioConfig),
+    #[serde(rename = "vhost")]
+    #[cfg(feature = "vhost-net")]
+    /// Vhost-net
+    Vhost(VirtioConfig),
+}
+
+impl Default for Backend {
+    fn default() -> Self {
+        Self::Virtio(VirtioConfig::default())
+    }
+}
+
+/// Virtio network config, working for virtio-net and vhost-net.
+#[derive(Clone, Debug, Default, Deserialize, PartialEq, Eq, Serialize)]
+pub struct VirtioConfig {
+    /// ID of the guest network interface.
+    pub iface_id: String,
+    /// Host level path for the guest network interface.
+    pub host_dev_name: String,
+    /// Rate Limiter for received packages.
+    pub rx_rate_limiter: Option<RateLimiterConfigInfo>,
+    /// Rate Limiter for transmitted packages.
+    pub tx_rate_limiter: Option<RateLimiterConfigInfo>,
+    /// Allow duplicate mac
+    pub allow_duplicate_mac: bool,
+}
+
+/// This struct represents the strongly typed equivalent of the json body from
+/// net iface related requests.
+/// This struct works with virtio-net devices and vhost-net devices.
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize, Default)]
+#[serde(deny_unknown_fields)]
+pub struct NetworkInterfaceConfig {
```

Comment:
Do we really need the convert from NetworkInterfaceConfig to NetworkInterfaceConfigInfo?

Reply:
Well, we can do this later.